### ISSUE TITLE
Paste right away without copying?

### ISSUE THREAD

**Issue by ilearnio** [p0deje/Maccy#14]
Right now when selecting an entry in clipboard history it gets copied to buffer. But to me it would make more sense if it would paste it right away without overwriting existing buffer. Maybe that's a matter of personal opinion/choice, but I wonder what do you think about it

**Comment by frank-f** [comment]
I would like an instant-paste feature as well, but I think for me it would be more convenient to have the item in the buffer as well. Maybe this can be done with different (configurable, like the hotkey) key combinations. I'd suggest return / option-return / command-return.

**Comment by p0deje** [comment]
I attempted to implement it as part of #7 (see my [comment](https://github.com/p0deje/Maccy/issues/7#issuecomment-507631022)) so it could be done using `Shift+Enter`, but had troubles dealing with macOS accessibility - sometimes the paste worked fine and sometimes it didn't work at all. I might just not know how to do it properly.

I'll be happy to accept PR which implements this feature though.

**Comment by saeedseyfi** [comment]
@p0deje here you can see how Clipy works:
https://github.com/Clipy/Clipy/blob/develop/Clipy/Sources/Services/PasteService.swift

**Comment by p0deje** [comment]
@saeedseyfi Thanks for the link, I managed to implement the paste.

However, the way it is implemented is pretty slow when we have 100+ records in history. The reason for slowness is somewhere deep inside the Cocoa framework and how it calculates menu size, so I'm not sure what to do about that.

@ilearnio @frank-f @saeedseyfi Are you ok reducing the history size from 999 to 99 to have this feature?

**Comment by ilearnio** [comment]
Sounds good to me as a user, but still hope that's a temporary solution. It's strange though that a framework is used to calculate the menu size when calling paste

**Comment by p0deje** [comment]
Nevermind. It seems like the performance is slow only during development, but when I build app for release, it is as fast as it should be.

I now have a working solution which I am going to test for some time before making a release.

**Comment by frank-f** [comment]
I thought about this a little more and instead of making this an option, maybe you can just create a second (configurable) hotkey. Cmd-Shift-c put's the selected entry in the clipboard, while Cmd-Shift-v puts it in the clipboard and pastes it right away. Would that be possible?

**Comment by p0deje** [comment]
The main reason I use the Option key is because it feels more native in my opinion. A bunch of menus in macOS alter its behavior when you press the Option key - some menu items change their name, some new items are added. Because of that, Cocoa has built-in mechanisms for altering menu items and it works really nice and as intuitive as default macOS apps.

I'd like to avoid introducing another global shortcut. However, I can add a configuration option which will turn on/off pasting by default:

```bash
defaults write org.p0deje.Maccy pasteByDefault true # this will paste by default and NOT paste when selected with the Option key pressed
defaults write org.p0deje.Maccy pasteByDefault false # this will NOT paste by default and paste when selected with the Option key pressed
```

Please react with :+1:/:-1: if you'd like to have that.

**Comment by scwood** [comment]
Does this feature currently work? Right now it's not doing anything for me

**Comment by p0deje** [comment]
Not released yet.

**Comment by p0deje** [comment]
This is now available in [0.4.0](https://github.com/p0deje/Maccy/releases/tag/0.4.0).
